src/actions.ts

```diff
@@ -467,6 +475,75 @@ export const actions: {[k: string]: QueryHandler} = {
 			matches: await tables.users.selectAll(['userid', 'banstate'])`WHERE ip = ${res.ip}`,
 		};
 	},
+	// oauth is broken into a few parts
+	// oauth/page - public-facing part
+	// oauth/api/page - api part (does the actual action)
+	async 'oauth/authorize'(params) {
+		if (!params.redirect_uri) {
+			throw new ActionError("No redirect_uri provided");
+		}
+		const clientInfo = await getOAuthClient(params.client_id);
+
+		this.response.setHeader('Content-Type', 'text/html');
+		try {
+			let content = await fs.readFile(
+				__dirname + "/../../src/public/oauth-authorize.html",
+				'utf-8'
+			);
+			// table keys are owner, clientName, id
+			// expects client, client_name, redirect_uri
+			content = content.replace(/\{\{client\}\}/g, clientInfo.client_title);
+			content = content.replace(/\{\{client_name\}\}/g, clientInfo.owner);
+			content = content.replace(/\{\{redirect_uri\}\}/g, params.redirect_uri);
```

Comment:
`params` is user input right? This would be vulnerable to XSS then, right? I would expect a comment explaining why that doesn't matter, or why it doesn't apply

Same applies to `clientInfo` stuff. Where does this data come from anyway? I don't see anywhere in the code where there is an insert in this table.

Reply:
Oh, yeah, that's true. Let me add a regex to sanitize that. 

There's no insert for the client info stuff at the moment, I said I'd just manually register people for now since Zarel wanted to do the design himself, iirc.

Reply:
OK, done. Sanitized the URL via a regex for links (stolen from PS main source.)